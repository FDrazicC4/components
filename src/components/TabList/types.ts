import { StyledParentComponent } from '../types';

export interface Props extends StyledParentComponent {
    /**
     * The number of pixels wide that the component should be before using mobile styles
     */
    mobileBreakpoint?: number,
};