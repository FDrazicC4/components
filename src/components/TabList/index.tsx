import React from 'react';
import MuiTabs from '@material-ui/core/Tabs';
import { useTabs } from '@components/Tabs/context';
import { Props } from './types';


/**
 * The `<TabList />` component provides a wrapper around the individual tabs.  It should exist as a child of the `<Tabs />` component.
 */
const TabList = React.forwardRef(({
    mobileBreakpoint = 600,
    ...otherProps
}: Props, ref: React.Ref<any>): React.ReactElement => {


    const [isMobile, setIsMobile] = React.useState(window && window.innerWidth <= mobileBreakpoint);
    const { value, onChange } = useTabs();


    const muiClasses = {
        root: 'brew-TabList',
        vertical: 'brew-TabList--vertical',
        flexContainer: 'brew-TabList__flexContainer',
        flexContainerVertical: 'brew-TabList__flexContainer--vertical',
        centered: 'brew-TabList__flexContainer--centered',
        scroller: 'brew-TabList__scroller',
        fixed: 'brew-TabList__scroller--fixed',
        scrollable: 'brew-TabList__scroller--scrollable',
        scrollButtons: 'brew-TabList__scrollButtons',
        scrollButtonsDesktop: 'brew-TabList__scrollButtons--desktop',
        indicator: 'brew-TabList__indicator',
    };


    React.useEffect(() => {

        const onWindowResize = () => {
            if (window.innerWidth <= mobileBreakpoint) {
                if (!isMobile) setIsMobile(true);
            }
            else {
                if (isMobile) setIsMobile(false);
            }
        }

        window.addEventListener('resize', onWindowResize);

        return () => {
            window.removeEventListener('resize', onWindowResize);
        }

    }, [isMobile, mobileBreakpoint]);


    return (
        <MuiTabs
            {...otherProps}
            variant={isMobile ? 'fullWidth' : 'scrollable'}
            classes={muiClasses}
            value={value}
            onChange={onChange}
            ref={ref}
        />
    );


});


export default TabList;
