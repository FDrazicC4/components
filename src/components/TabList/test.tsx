import React from 'react';
import renderer from 'react-test-renderer';
import TabList from '.';


describe('TabList', () => {

    it('Renders correctly with defaults', () => {
        const tree = renderer
            .create(<TabList />)
            .toJSON();
        expect(tree).toMatchSnapshot();
    });

});
