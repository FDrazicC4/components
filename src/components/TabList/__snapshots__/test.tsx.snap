// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabList Renders correctly with defaults 1`] = `
<div
  className="MuiTabs-root brew-TabList"
>
  <div
    className="MuiTabs-scrollable brew-TabList__scroller--scrollable"
    style={
      Object {
        "height": 99,
        "overflow": "scroll",
        "position": "absolute",
        "top": -9999,
        "width": 99,
      }
    }
  />
  <div
    className="MuiTabs-scroller brew-TabList__scroller MuiTabs-scrollable brew-TabList__scroller--scrollable"
    onScroll={[Function]}
    style={
      Object {
        "marginBottom": null,
        "overflow": "hidden",
      }
    }
  >
    <div
      className="MuiTabs-flexContainer brew-TabList__flexContainer"
      role="tablist"
    />
  </div>
</div>
`;
